### PR TITLE
fix rollup apply join

### PR DESCRIPTION
**Pull request by nevermore3**
## What type of PR is this?
- [ ] bug
- [ ] feature
- [ ] enhancement

#### Issue(s) number: 
close https://github.com/vesoft-inc/nebula/issues/5209
## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

File: src/graph/executor/Executor.h [hunk 1]
```diff
@@ -17,6 +17,98 @@ namespace nebula {
...
+          seed ^= std::hash<nebula::Value>()(v);
+        }
+      }
+    }
+    DLOG(ERROR) << "hashValue : " << wrapper.toString() << " values : " << seed;
+    return seed;
+  }
+};
+
+struct WrapperEqual {
```
Comment (yixinglu) [inline src/graph/executor/Executor.h:66]:
```
namespace std {
 template <>
 struct hash<ListWrapper> {
  size_t operator()(const ListWrapper& l) {}
 };
 template <>
  struct equal_to<ListWrapper> {
    int operator()(const ListWrapper&) {}
  };
}
```

File: tests/tck/features/match/PathExpr.feature [hunk 2]
```diff
@@ -447,3 +447,34 @@ Feature: Basic match
...
       | v |
+
+  Scenario: different edge type in two pattern when apply join
+    When executing query:
+      """
+      MATCH (v:player{name: 'Tim Duncan'})-[e:like*2]->(n)
+        RETURN ()-[e:like*2]->(n)
+      """
+    Then the result should be, in any order:
+      | ()-[e:like*2]->(n) = ()-[e:like*2]->(n)                                                                                                                                                                                                                                                                                                   |
```
Comment (yixinglu) [inline tests/tck/features/match/PathExpr.feature:458]: why the column name is different with the query?

File: tests/tck/features/match/PathExpr.feature [hunk 3]
```diff
@@ -447,3 +447,34 @@ Feature: Basic match
...
+      | [<("Tim Duncan" :player{age: 42, name: "Tim Duncan"} :bachelor{name: "Tim Duncan", speciality: "psychology"})-[:like@0 {likeness: 95}]->("Tony Parker" :player{age: 36, name: "Tony Parker"})-[:like@0 {likeness: 90}]->("LaMarcus Aldridge" :player{age: 33, name: "LaMarcus Aldridge"})>]                                               |
+      | [<("Tim Duncan" :player{age: 42, name: "Tim Duncan"} :bachelor{name: "Tim Duncan", speciality: "psychology"})-[:like@0 {likeness: 95}]->("Tony Parker" :player{age: 36, name: "Tony Parker"})-[:like@0 {likeness: 95}]->("Manu Ginobili" :player{age: 41, name: "Manu Ginobili"})>]                                                       |
+      | [<("Tim Duncan" :player{age: 42, name: "Tim Duncan"} :bachelor{name: "Tim Duncan", speciality: "psychology"})<-[:like@0 {likeness: 95}]-("Tony Parker" :player{age: 36, name: "Tony Parker"})<-[:like@0 {likeness: 95}]-("Tim Duncan" :player{age: 42, name: "Tim Duncan"} :bachelor{name: "Tim Duncan", speciality: "psychology"})>]     |
+    When executing query:
+      """
+      MATCH (v:player{name: 'Tim Duncan'})-[e:like]->(n)
+        RETURN ()-[e:like]->(n)
+      """
```
Comment (yixinglu) [inline tests/tck/features/match/PathExpr.feature:467]: more tests: bi-direction

```
      MATCH (v:player{name: 'Tim Duncan'})-[e:like]-(n)
        RETURN ()-[e:like]-(n)
```

File: tests/tck/features/match/PathExpr.feature (hunk 2 above)
Comment (nevermore3) [reply tests/tck/features/match/PathExpr.feature:458]: fixed

File: tests/tck/features/match/PathExpr.feature (hunk 3 above)
Comment (nevermore3) [reply tests/tck/features/match/PathExpr.feature:467]: done

File: src/graph/executor/Executor.h (hunk 1 above)
Comment (nevermore3) [reply src/graph/executor/Executor.h]: done

**Pull request closed without merging**